### ISSUE TITLE
software problem or mounting pot problem?

### ISSUE THREAD

**Issue by pipes80-collab** [FreeJoy-Team/FreeJoyConfiguratorQt#11]
I managed to convert the g29 pedal board to usb, by connecting the potentiometers directly to the stm32 board, before closing the top cover I tested it and I didn't have that kind of oscillation that you see in the video of the axis scroll bar.
any opinion? I hope I was clear in explaining the problem, in game I have no problems, at least for what little I was able to try

https://user-images.githubusercontent.com/56699284/114300313-04365500-9ac0-11eb-881f-7c18e7082b3d.mp4

**Comment by Reksotiv** [comment]
why do you have a packet speed of 1ms? these are debug packages only for configurator and should be 16 ms

**Comment by pipes80-collab** [comment]
I had no major problems before the change even at 1 ms, do you think that is the problem?  Wouldn't it have better speed performance at 1ms?

**Comment by Reksotiv** [comment]
> Wouldn't it have better speed performance at 1ms?

No. These are debug packages for configurator only. Configurator is refreshed no more than once every 17ms. 

> do you think that is the problem?

I don't see a problem in the video if you mean noise

**Comment by pipes80-collab** [comment]
note that holding the pedal down, the bar has a kind of slight wobble and the values do not remain stable and fixed, at least they were in the way the potentiometers were originally connected

**Comment by Reksotiv** [comment]
it is noise. I already answered https://github.com/FreeJoy-Team/FreeJoyConfiguratorQt/issues/5

**Comment by Reksotiv** [comment]
https://en.wikipedia.org/wiki/Noise_(electronics)

**Comment by pipes80-collab** [comment]
It's normal but less noise in original configuration...I take the question as closed... thanks for help

**Comment by pipes80-collab** [comment]
Sorry if I have reopen, it can be help for noise this solution?
https://forum.arduino.cc/index.php?topic=371953.30

**Comment by pipes80-collab** [comment]
Or this solution https://forum.arduino.cc/index.php?topic=363175.0

**Comment by Reksotiv** [comment]
this can help. [Logitech G25](https://forum.simracing.su/uploads/monthly_2020_06/319461467_schematicG25.jpg.b873c567f0811aa92576d4f4f5d8b929.jpg)

**Comment by vostrenkov** [comment]
Capacitors and digital filters should give you quite similar results. Have you tried to add filter levels for axes?
Capacitor or digital filter should be applied to every axis that you want to be smoother.

`the problem did not appear before closing and gluing the board with hot glue`
this seem to be some kind of shielding issue. Keep your wires as short as possible, use twisted pair cable and prevent gnd and pwr circuits from looping

**Comment by pipes80-collab** [comment]
> Capacitors and digital filters should give you quite similar results. Have you tried to add filter levels for axes?
> Capacitor or digital filter should be applied to every axis that you want to be smoother.
> 
> `the problem did not appear before closing and gluing the board with hot glue`
> this seem to be some kind of shielding issue. Keep your wires as short as possible, use twisted pair cable and prevent gnd and pwr circuits from looping

the capacitors must be placed on each axis ground, which capacitors to use? polymers or multilayer or other?
I will have made the cables 30 cm long to be able to get to a comfortable point inside the pedal box case, are they cables with silicone sheath, maybe that holding them close, are 3 wires for potentiometer can create electromagnetic interference?

**Comment by Reksotiv** [comment]
[Logitech G25](https://forum.simracing.su/uploads/monthly_2020_06/1315449104_.jpg.27a8fd8696b26d15a943356dbfe3fc0c.jpg)

**Comment by vostrenkov** [comment]
Ceramic capacitors should be fine. Value is dependent of the resistance of your pots and can be counted by formula for low pass filter

**Comment by pipes80-collab** [comment]
Have 10k pot

**Comment by vostrenkov** [comment]
> Have 10k pot
> 

In this case 0.1-1uF capacitor should be ok for you. Note that filter will work in different way at different positions of pot because cut frequency is determined by formula f=1/(2*pi*R*C)

**Comment by pipes80-collab** [comment]
It's good thing use ferrite for EMI?

**Comment by pipes80-collab** [comment]
is it possible that this problem is due to the fact that each potentiometer is connected to a different 3.3 volt line?
